pkg/utils/common_test.go

```diff
@@ -27,4 +27,29 @@ var _ = Describe("Unit tests for the common utils", func() {
 		present := ContainsString(testStrings, "foobar")
 		Expect(present).To(BeFalse())
 	})
+
+	It("Test the JoinURL function with different inputs", func() {
+		testCases := []struct {
+			baseURL     string
+			relativeURL string
+			expected    string
+		}{
+			{"https://www.example.com/", "/test/path/", "https://www.example.com/test/path"},
+			{"https://www.example.com", "/test/path/", "https://www.example.com/test/path"},
+			{"https://www.example.com", "test/path/", "https://www.example.com/test/path"},
+			{"https://www.example.com/", "test/path/", "https://www.example.com/test/path"},
+			{"https://www.example.com", "test/path", "https://www.example.com/test/path"},
+			{"https://www.example.com/", "/test/path", "https://www.example.com/test/path"},
+			{"https://www.example.com/", "test/path", "https://www.example.com/test/path"},
+			{"https://www.example.com", "/test/path", "https://www.example.com/test/path"},
+			{"https://www.example.com", "", "https://www.example.com"},
+			{"https://www.example.com/", "", "https://www.example.com/"},
+			{"https://www.example.com/", "/", "https://www.example.com/"},
```

Comment:
Can we add a unit test that doesn't contain the URL schema (without `https://` at the front)? This is because the container repository will not have the schema defined and I have a feeling that `url.Parse` requires the schema to be defined.

Like 
```
{"www.example.com", "/test/path", "www.example.com/test/path"},
{"example.com/", "/test/path", "example.com/test/path"},
```

Reply:
Done. Added

Reply:
Added more test cases

Reply:
Thanks for verifying.